### PR TITLE
Initial Xtensa support

### PR DESCRIPTION
**Pull request by MabezDev**
This PR adds the bare minimum Xtensa support to allow our [Xtensa-enabled Rust fork](https://github.com/esp-rs/rust) to compile on stable 1.65, as we were running into https://github.com/rust-lang/rust/issues/102290 in our testing.

### CODE REVIEW

**Comment by philipc** [comment]
Please add this architecture to the [`elf_any` test](https://github.com/gimli-rs/object/blob/389c43a7567df94bbb31c697f79ed473523a7089/tests/round_trip/mod.rs#L231). #430 is a PR that did this for another architecture.

**Comment by MabezDev** [comment]
Thanks for the link to #430, was most helpful! I've gone ahead and implemented the relevant relocations for Xtensa and added the Xtensa arch to the test case as you requested.

File: src/write/elf/object.rs [hunk 1]
```diff
@@ -673,8 +675,7 @@ impl<'a> Object<'a> {
                         },
                         Architecture::Sparc64 => match (reloc.kind, reloc.encoding, reloc.size) {
                             // TODO: use R_SPARC_32/R_SPARC_64 if aligned.
-                            (RelocationKind::Absolute, _, 32) => elf::R_SPARC_UA32,
-                            (RelocationKind::Absolute, _, 64) => elf::R_SPARC_UA64,
+                            (RelocationKind::Absolute, _, 32) => elf::R_XTENSA_32,
```
Comment (philipc) [inline src/write/elf/object.rs:678]: I don't think you meant to change this.

File: src/write/elf/object.rs [hunk 2]
```diff
@@ -687,6 +688,16 @@ impl<'a> Object<'a> {
                                 return Err(Error(format!("unimplemented relocation {:?}", reloc)));
                             }
                         }
+                        Architecture::Xtensa => match (reloc.kind, reloc.encoding, reloc.size) {
+                            (RelocationKind::Absolute, _, 32) => elf::R_XTENSA,
```
Comment (philipc) [inline src/write/elf/object.rs:692]: Should this be `R_XTENSA_32`?

**Review by philipc — CHANGES_REQUESTED** [review]

**Review by philipc — CHANGES_REQUESTED** [review]
Please run `cargo fmt`.

**Review by philipc — APPROVED** [review]
Thanks!

**Comment by MabezDev** [comment]
Thanks, @philipc! Are there any plans to cut a new release anytime soon?

**Comment by philipc** [comment]
I'd like to do a release soon. However, there's a bit of XCOFF work in progress currently that I'd like to see polished a bit more first. I may consider doing a release with the XCOFF feature disabled somehow.

**Pull request merged**